sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayAsyncClient.java

```diff
@@ -64,34 +98,53 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser) {
-        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, context));
+        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, null, context));
     }
 
     /**
      * Creates a new CommunicationRelayConfiguration with response.
      *
+     * @param routeType The specified CommunicationRelayConfigurationRequestRouteType for the relay request
      * @return The created Communication Relay Configuration.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse() {
-        return withContext(context -> getRelayConfigurationWithResponse(null, context));
+    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationRelayConfigurationRequestRouteType routeType) {
+        return withContext(context -> getRelayConfigurationWithResponse(null, routeType, context));
+    }
+
+    /**
+     * Creates a new CommunicationRelayConfiguration with response.
+     *
+     * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @param routeType The specified CommunicationRelayConfigurationRequestRouteType for the relay request
+     * @return The created Communication Relay Configuration.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser, CommunicationRelayConfigurationRequestRouteType routeType) {
+        return withContext(context -> getRelayConfigurationWithResponse(communicationUser, routeType, context));
     }
 
     /**
      * Creates a new CommunicationRelayConfiguration with response.
      *
      * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @param routeType The specified CommunicationRelayConfigurationRequestRouteType for the relay request
      * @param context A {@link Context} representing the request context.
      * @return The created Communication Relay Configuration.
      */
-    Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser, Context context) {
+    Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser, CommunicationRelayConfigurationRequestRouteType routeType, Context context) {
```

Comment:
I see that the javadocs for all `getRelayConfiguration*` APIs suggest that it is creating a configuration. I am now curious whether this API is actually just getting a config or is it creating one.